client/src/pages/App/Parents/AddRecommendationModal/AddRecommendationModal.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  Button,
+  Form,
+  FormField,
+  Header,
+  Modal,
+  ModalContent,
+  ModalDescription,
+  ModalHeader,
+} from "semantic-ui-react";
+import { useMutation } from "react-query";
+import { FieldValues, useForm } from "react-hook-form";
+
+import { ModalAddReviewProps } from "./RecommendationModalInterfaces";
+import { addRecommendation } from "./recommendationServices";
+
+const AddRecommendationModal = ({
+  isOpen,
+  setIsOpen,
+  parentId,
+  babysitterId,
+  babysitterName,
+}: ModalAddReviewProps) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { isSubmitting },
+  } = useForm();
+
+  const { mutate } = useMutation({
+    mutationKey: ["addRecommendation"],
+    mutationFn: addRecommendation,
+    onError: (error: unknown) => {
+      reset();
+      console.log(error);
+    },
+  });
+
+  const onSubmit = async (data: FieldValues) => {
+    mutate({ data, babysitterId, parentId });
+  };
+
+  return (
+    <Modal
+      closeIcon
+      onClose={() => setIsOpen(false)}
+      onOpen={() => setIsOpen(true)}
+      open={isOpen}
+    >
+      <ModalHeader>{`Post a review about ${babysitterName}`}</ModalHeader>
+      <ModalContent>
+        <ModalDescription>
+          <Header>{`Post a review about ${babysitterName}`}</Header>
+          <Form onSubmit={handleSubmit(onSubmit)}>
+            <FormField>
+              <textarea
+                placeholder="Add your review"
+                {...register("recommendationText", {
+                  required: "please provide an review",
+                })}
+              />
+            </FormField>
+
+            <FormField>
+              <label>Rating</label>
+              <input
```

Comment:
I don't like this.
Semantic has a rating starts - use that
https://react.semantic-ui.com/modules/rating/